spacemesh/v2alpha1/tx.proto

```diff
@@ -96,26 +97,26 @@ message TransactionResult {
 }
 
 message TransactionStreamRequest {
-  uint32 start_layer = 1;
-  uint32 end_layer = 2;
-  repeated bytes id = 3;
-  string address = 4;
-  bool watch = 5;
+  uint32 start_layer = 1; // Starting layer for transaction streaming.
+  uint32 end_layer = 2; // Ending layer for transaction streaming.
+  repeated bytes txid = 3; // Filter: specific transaction IDs.
+  string address = 4; // Filter: address involved in transaction.
+  bool watch = 5; // Continue watching for new transactions.
 }
 
 service TransactionStreamService {
   rpc Stream(TransactionStreamRequest) returns (stream Transaction);
 }
 
 message TransactionRequest {
-  uint32 start_layer = 1;
-  uint32 end_layer = 2;
-  repeated bytes id = 3;
-  string address = 4;
-  uint64 offset = 5; // adjusts the starting point for data
-  uint64 limit = 6; // specifies max number of items to fetch
-  bool include_state = 7;
-  bool include_result = 8;
+  optional uint32 start_layer = 1; // Starting layer for transactions.
+  optional uint32 end_layer = 2; // Ending layer for transactions.
+  repeated bytes txid = 3; // Filter: specific transaction IDs to filter.
+  optional string address = 4; // Filter: address involved in transaction.
+  uint64 offset = 5; // Adjusts the starting point for data retrieval.
```

Comment:
Required or optional?

Reply:
offset and limit is always required - just like in rewards, atxs, layers etc